Sources/PackageLoading/TargetSourcesBuilder.swift

```diff
@@ -104,33 +109,14 @@ public struct TargetSourcesBuilder {
             }
         }
 
-        let compilePaths = pathToRule.filter{ $0.value == .compile }.map{ $0.key }
-        let sources = Sources(paths: compilePaths, root: targetPath)
-
-        let resources: [Resource] = pathToRule.compactMap {
-            switch $0.value {
-            case .compile, .none, .modulemap:
-                return nil
-            case .processResource:
-                return Resource(rule: .process, path: $0.key)
-            case .copy:
-                return Resource(rule: .copy, path: $0.key)
-            }
-        }
-
-        // Diagnose conflicting resources.
-        let duplicateResources = resources
-            .spm_findDuplicateElements(by: \.path.basename)
-        for resources in duplicateResources {
-            let filename = resources[0].path.basename
-            diags.emit(.conflictingResource(filename: filename, targetName: target.name))
-
-            for resource in resources {
-                let relativePath = resource.path.relative(to: targetPath)
-                diags.emit(.fileReference(path: relativePath))
-            }
-        }
+        let compilePaths = pathToRule.lazy.filter({ $0.value.rule == .compile }).map({ $0.key })
```

Comment:
Is lazy useful here? Not sure if compiler can automatically optimize this and automatically add lazy?

Reply:
I have no idea. I tried checking the difference in Godbolt, but I'm not an assembler wizard. Do you want me to remove it?